### DISCUSSION TITLE
model selection via Bayes factors in `Turing.jl`

### DISCUSSION THREAD

**Discussion by SimonEnsemble** [TuringLang/Turing.jl#1864] [Q&A]
interested in model selection via Bayes factors.

my understanding is that `Turing.jl` currently provides no functionality for this.

[this blog post](http://jakevdp.github.io/blog/2015/08/07/frequentism-and-bayesianism-5-model-selection/#Bayesian-Model-Selection:-the-Odds-Ratio) shows how we can numerically integrate the posterior with quadrature to obtain the evidence, provided (i) there aren't too many params to make this integral high-dimensional and (ii) we can evaluate the posterior at a given value of the params.

what do you think of the following approach to computing Bayes' factors, using `Turing.jl` (my models only have 2-3 params :smile:).
* sample to obtain a `chain` as in the tutorials of `Turing.jl`.
* use kernel density estimation method to approximate the posterior distribution over param space
* feed kernel density estimate into a numerical quadrature routine to compute evidence.

**Comment by SimonEnsemble** [comment]
if this approach makes sense, happy to write a short tutorial demonstrating this, for `Turing.jl`.

**Reply by gretarsson** [reply]
Did you make the tutorial?

**Reply by SimonEnsemble** [reply]
no---no one responded so I assumed no interest. 😭 
could do on a 'rainy day' this summer if this would be of interest, though!

**Reply by torfjelde** [reply]
Completely missed this 😬 But indeed seems interesting! Think we'd be very happy to have a tutorial like this

@yebai
